client/packages/common/src/intl/currency/currency.ts

```diff
@@ -67,80 +67,120 @@ export const format: currency.Format = (
-    separator: '.',
-    decimal: ',',
-    precision: 2,
-    pattern: '# !',
-    negativePattern: '-# !',
-    format,
-  },
-  ar: {
-    symbol: 'ر.ق.',
-    separator: ',',
-    decimal: '.',
-    precision: 2,
-    pattern: '!#',
-    negativePattern: '-!#',
-    format,
-  },
-  es: {
-    symbol: '$',
-    separator: ',',
-    decimal: '.',
-    precision: 2,
-    pattern: '!#',
-    negativePattern: '-!#',
-    format,
-  },
-  tet: {
-    symbol: '$',
-    separator: ',',
-    decimal: '.',
-    precision: 2,
-    pattern: '!#',
-    negativePattern: '-!#',
-    format,
-  },
-  ru: {
-    symbol: '₽',
-    separator: '.',
-    decimal: ',',
-    precision: 2,
-    pattern: '# !',
-    negativePattern: '-# !',
-    format,
-  },
+export type Currencies =
+  | 'USD'
+  | 'EUR'
+  | 'NZD'
+  | 'DJF'
+  | 'QAR'
+  | 'RUB'
+  | 'SSP'
+  | 'PGK'
+  | 'COP';
+
+export const currencyOptions = (code?: Currencies) => {
+  switch (code) {
+    case 'USD':
+    case 'NZD':
+      return {
+        symbol: '$',
+        separator: ',',
+        decimal: '.',
+        precision: 2,
+        pattern: '!#',
+        negativePattern: '-!#',
+        format,
+      };
+    case 'EUR':
+      return {
+        symbol: '€',
+        separator: '.',
+        decimal: ',',
+        precision: 2,
+        pattern: '# !',
+        negativePattern: '-# !',
+        format,
+      };
+    case 'DJF':
+      return {
+        symbol: 'DJF',
+        separator: ',',
+        decimal: '.',
+        precision: 2,
+        pattern: '# !',
+        negativePattern: '-# !',
+        format,
+      };
+    case 'QAR':
+      return {
+        symbol: 'ر.ق.',
+        separator: ',',
+        decimal: '.',
+        precision: 2,
+        pattern: '!#',
+        negativePattern: '-!#',
+        format,
+      };
+    case 'RUB':
+      return {
+        symbol: '₽',
+        separator: '.',
+        decimal: ',',
+        precision: 2,
+        pattern: '# !',
+        negativePattern: '-# !',
+        format,
+      };
+    case 'SSP': {
+      return {
+        symbol: 'SSP',
+        pattern: '# !',
+        negativePattern: '-# !',
+        format,
+      };
+    }
+    case 'PGK': {
+      return {
+        symbol: 'K',
+        separator: '.',
+        decimal: ',',
+        precision: 2,
+        pattern: '# !',
+        negativePattern: '-# !',
+        format,
+      };
+    }
+    case 'COP': {
+      return {
+        symbol: '$',
+        pattern: '# !',
+        negativePattern: '-# !',
+        format,
+      };
+    }
+    default:
+      return {
+        symbol: '$',
+        separator: ',',
+        decimal: '.',
+        precision: 2,
+        pattern: '!#',
+        negativePattern: '-!#',
+        format,
+      };
+  }
 };
 
-export const useCurrency = (dp?: number) => {
-  const { currentLanguage: language } = useIntlUtils();
-  const options = currencyOptions[language];
+export const useCurrency = (dp?: number, code?: Currencies) => {
```

Comment:
I can find no usage of `useCurrency` or `useFormatCurrency` that actually use the dp parameter, so I'm tempted to simply drop it.

if you think we should retain `dp` as a parameter it would be nicer if this was an object, then we don't have to pass `undefined` for the first param when you only want to specify the second..

```suggestion
export const useCurrency = ({dp, code}: {dp?: number, code?: Currencies}) => {
```